### PR TITLE
Added src, bin and documentation for inserting a nthash in a LDAP attrib...

### PR DESCRIPTION
**Pull request by fdurand**
DLL to implement Active Directory Password Filters to intercept clear text password and update ldap user attribute with the nthash calculated based on the password.

### CODE REVIEW

File: addons/nthash_AD_attribute/README.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+LDAP Display Name: ntPassword
+Unique x500 Object ID: 1.3.6.1.4.1.2964.2.1
+Description: NTHASH Password
+
+Syntax: Case Sensitive String
+```
+
+And validate
+
+Now go in Classes and add ntPassword attribute as an optional attribute for the user classe (Do the same for computer).
```
Comment (julsemaan) [inline addons/nthash_AD_attribute/README.md]: user class

File: addons/nthash_AD_attribute/README.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+# Security
+
+Because nthash is considered unsecured we must add security on Active Directory in order to make only one user to be able to write and read this attribute.
```
Comment (julsemaan) [inline addons/nthash_AD_attribute/README.md]: *to make only one user able

File: addons/nthash_AD_attribute/README.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+
+First create a new user in active directory (it will be use to update the nthash attribute and to connect to AD from freeradius) like PacketFenceAdmin.
```
Comment (julsemaan) [inline addons/nthash_AD_attribute/README.md]: - it will be used

Comment (julsemaan) [inline addons/nthash_AD_attribute/README.md]: *Active Directory

File: addons/nthash_AD_attribute/README.md [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+(The dn of this user is CN=PacketFenceAdmin,CN=Users,DC=inverse,dc=inc)
+
+Now in Active Directory Schema mmc console, right click on User class (you will have to do the same thing for Computer classe) then go on Default Security, Advanced and add these permissions:
```
Comment (julsemaan) [inline addons/nthash_AD_attribute/README.md]: *Now in the

Comment (julsemaan) [inline addons/nthash_AD_attribute/README.md]: *Computer class

File: addons/nthash_AD_attribute/README.md [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+
+```
+Allow PacketFenceAdmin to Read ntPassword and write ntPassword
+Allow Administrator to Read ntPassword and write ntPassword
+Deny everyone to read ntPassword and write ntPassword
```
Comment (julsemaan) [inline addons/nthash_AD_attribute/README.md]: should read be written Read like the other ones ?

File: addons/nthash_AD_attribute/README.md [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+
+Launch regedit and edit the HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Lsa\Notification Packages then add HashingPasswordFilter at the end of the list.
+
+Then restart the server.
+
+# First test
+
+Once the server reboot, go in Active Directory Users and Computers and change a password of a user (Robert).
+Then open c:\ProgramData\HashingPasswordFilter.log, you will see that Robert´s password has been changed.
+Then now launch adsiedit.msc and edit the robert user, you will see the ntPassword attribute that contain the password´s NTHASH.
```
Comment (julsemaan) [inline addons/nthash_AD_attribute/README.md]: *that contains

File: addons/nthash_AD_attribute/README.md [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+
+
+# Freeradius configuration
+
+First we have to configure ldap connection, in /usr/local/pf/raddb/modules edit ldap file and add ad_user and ad_computer configuration:
```
Comment (julsemaan) [inline addons/nthash_AD_attribute/README.md]: *configure the ldap

**Comment by fdurand** [comment]
Can it be merge ?

**Pull request merged**